### ISSUE TITLE
Suggest usage of `array.as_slice()` (instead of `&array[..]`) and `array.as_mut_slice()`

### ISSUE THREAD

**Issue by Luro02** [rust-lang/rust-clippy#7633]
### What it does

Lints for example this code:

```rust
let array: [u8; 4] = [1, 2, 3, 4];
let slice: &[u8] = &array[..];
```
and suggests [`array::as_slice`](https://doc.rust-lang.org/std/primitive.array.html#method.as_slice):

```rust
let array: [u8; 4] = [1, 2, 3, 4];
let slice: &[u8] = array.as_slice();
```

(same should be suggested for `&mut array[..]`, which can be replaced with `array.as_mut_slice()`)

### Categories (optional)

- Kind: `clippy::pedantic` or `clippy::style`

### What is the advantage of the recommended code over the original code?

For example:
- easier to read for people new to rust (not very obvious what `[..]` does)
- there must be a reason why `array::as_slice` exists? (see rust-lang/rust#76120)

### Drawbacks

- might have a lot of false positives or not much benefit in using the function

### Possibly related issues:
- #728
- #5598 

### Stabilization
At the time of writing this issue, the function is still unstable, but with the merging of rust-lang/rust#88353, this feature will be stabilized. (final-comment-period, so will be merged in the next few days)

**Comment by mikerite** [comment]
Not sure about this one. We currently suggest `for x in &y { .. }` over `for x in y.iter() { .. }` - [explicit_iter_loop](https://rust-lang.github.io/rust-clippy/master/index.html#explicit_iter_loop). This seems like it's going to the opposite direction in terms of style. Also we don't current recommend `Vec::as_slice` over slicing although we could have for years.

**Comment by Dmitry-Borodin** [comment]
Should this issue be implemented or it's doubtable? 
I'm looking for a good first issue, and I would like to understand if this should be implemented as in original message and under which category it's better to place.

**Comment by giraffate** [comment]
IMO it looks like a good idea to implement this, but the `restriction` category might be better.
